mod/state-transition/pkg/core/state_processor.go

```diff
@@ -94,6 +94,11 @@ type StateProcessor[
 	// processingGenesis allows initializing correctly
 	// eth1 deposit index upon genesis
 	processingGenesis bool
+
+	// prevEpochValidators tracks the set of validators active during
+	// previous epoch. This is useful at the turn of the epoch to send to
+	// consensus only the diffs rather than the full updated validator set.
+	prevEpochValidators []*transition.ValidatorUpdate
```

Comment:
note that we only keep active validators here, i.e. those whose effective balance is larger than ejection one.
I do not expect this list to be beefy, although we should probably track its size and decide whether this is the best way to calculate diffs.

A less memory consuming approach is to track chances wherever we do them (e.g.  when updating effective balances, or creating validators or removing it). I prefer current approach is it is more easily verifiable for correctness for me, but we should measure how much of a performance improvement these two alternatives bring